odoo/models.py

```diff
@@ -6005,6 +6006,18 @@ def process(res):
                 ))
 
         if onchange in ("1", "true"):
+            # before calling on change ensure float are properly rounded
+            # as expected in configuration
+            field = self._fields[field_name]
+            if field.type == 'float' and field._digits:
+                setattr(
```

Comment:
@odoo/rd-security I think there are new checks by CI since I've stat this PR.

How should I rewrite this code to avoid security issue ? which kind of risk should we mitigate here?